internal/hostnetwork/ovs_bridge.go

```diff
@@ -0,0 +1,356 @@
+			libovsclient.WithTable(&Interface{}),
+		),
+	)
+	if err != nil {
+		return fmt.Errorf("failed to setup monitor: %w", err)
+	}
+
+	bridgeUUID, err := ensureBridge(ctx, ovs, bridgeName)
+	if err != nil {
+		return fmt.Errorf("failed to ensure OVS bridge %q exists: %w", bridgeName, err)
+	}
+
+	// Create the bridge management port (internal port) so the bridge appears as a Linux interface
+	if err := ensureInternalPortForBridge(ctx, ovs, bridgeUUID, bridgeName); err != nil {
+		return fmt.Errorf("failed to create internal port for bridge %q: %w", bridgeName, err)
+	}
+
+	if err := ensurePortAttachedToBridge(ctx, ovs, bridgeUUID, ifaceName); err != nil {
+		return fmt.Errorf("failed to attach veth %q to bridge %q: %w", ifaceName, bridgeName, err)
+	}
+
+	// Bring the OVS bridge interface UP so it can be used as a master for macvlan/ipvlan
+	bridge, err := netlink.LinkByName(bridgeName)
+	if err != nil {
+		return fmt.Errorf("failed to get OVS bridge %q interface: %w", bridgeName, err)
+	}
+	if err := netlink.LinkSetUp(bridge); err != nil {
+		return fmt.Errorf("failed to bring OVS bridge %q UP: %w", bridgeName, err)
+	}
+	slog.Debug("OVS bridge interface is UP", "name", bridgeName)
+
+	return nil
+}
+
+func ensureBridge(ctx context.Context, ovs libovsclient.Client, bridgeName string) (string, error) {
+	br := &Bridge{Name: bridgeName}
+	err := ovs.Get(ctx, br)
+	if err == nil {
+		return br.UUID, nil
+	}
+
+	namedUUID := "new_bridge"
+	br = &Bridge{
+		UUID:        namedUUID,
+		Name:        bridgeName,
+		ExternalIds: map[string]string{"created-by": "openperouter"},
+	}
+
+	insertOp, err := ovs.Create(br)
+	if err != nil {
+		return "", fmt.Errorf("failed to create bridge insert operation: %w", err)
+	}
+
+	ovsRow := &OpenVSwitch{}
+	mutateOp, err := ovs.WhereCache(func(*OpenVSwitch) bool { return true }).
+		Mutate(ovsRow, model.Mutation{
+			Field:   &ovsRow.Bridges,
+			Mutator: ovsdb.MutateOperationInsert,
+			Value:   []string{namedUUID},
+		})
+	if err != nil {
+		return "", fmt.Errorf("failed to create mutate operation: %w", err)
+	}
+
+	operations := append(insertOp, mutateOp...)
+	reply, err := ovs.Transact(ctx, operations...)
+	if err != nil {
+		return "", fmt.Errorf("transaction failed: %w", err)
+	}
+
+	_, err = ovsdb.CheckOperationResults(reply, operations)
+	if err != nil {
+		return "", fmt.Errorf("operation failed: %w", err)
+	}
+
+	realUUID := reply[0].UUID.GoUUID
+	slog.Debug("created OVS bridge", "name", bridgeName, "UUID", realUUID)
+
+	return realUUID, nil
+}
+
+func ensurePortAttachedToBridge(ctx context.Context, ovs libovsclient.Client, bridgeUUID, interfaceName string) error {
+	iface := &Interface{Name: interfaceName}
+	interfaceUUID := ""
+	err := ovs.Get(ctx, iface)
+	if err == nil {
+		interfaceUUID = iface.UUID
+		slog.Debug("interface already exists", "name", interfaceName, "UUID", interfaceUUID)
+	}
+
+	port := &Port{Name: interfaceName}
+	portUUID := ""
+	err = ovs.Get(ctx, port)
+	if err == nil {
```

Comment:
is there a specific case where err is not nil because of a real error instead of port not existing? Can we handle that? Same for the other ignored errors

Reply:
how about we flip the if handling the error? 

if err != nil && !errors.Is() {
   early return
}
happy path